prefork/prefork.go

```diff
@@ -110,30 +144,91 @@ func (p *Prefork) setTCPListenerFiles(addr string) error {
 	return nil
 }
 
-func (p *Prefork) prefork(addr string) error {
-	chErr := make(chan error, 1)
+func (p *Prefork) doCommand() (*exec.Cmd, error) {
+	/* #nosec G204 */
+	cmd := exec.Command(os.Args[0], append(os.Args[1:], preforkChildFlag)...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.ExtraFiles = p.files
+	return cmd, cmd.Start()
+}
 
+func (p *Prefork) prefork(addr string) (err error) {
 	if !p.Reuseport {
-		if err := p.setTCPListenerFiles(addr); err != nil {
-			return err
+		if err = p.setTCPListenerFiles(addr); err != nil {
+			return
 		}
 
-		defer p.ln.Close()
+		// defer for closing the net.Listener opened by setTCPListenerFiles.
+		defer func() {
+			e := p.ln.Close()
```

Comment:
Should the listener from reuseport not be closed?

Reply:
`p.ln` doesn't even get assigned an instance of `net.Listener` under reuseport mode.

Reply:
Is the comment not good enough?